scripts/win-ci-logs-collector.sh

```diff
@@ -0,0 +1,165 @@
+#!/bin/bash
+# Copyright 2020 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+if [ "$#" -ne 3 ]; then
+    echo "Unexpected number of parameters $#."
+    echo "Usage: win-ci-logs-collector.sh masterIp sshPrivateKeyPath aksEngineOutputPath"
+    exit 1
+fi
+
+MASTER_IP=${1}
+TEMP_AKS_PATH=${2}
+SSH_KEY=${3}
+USER="azureuser"
+SSH_OPTS="-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null"
+
+# Check for ssh key in path
+if [ -z "$SSH_KEY" ]
+then
+    echo "ssh key not found. Defaulting to .ssh/id_rsa."
+    SSH_KEY="${HOME}/.ssh/id_rsa"
+fi
+
+if [ ! -f "$SSH_KEY" ]
+then
+    echo "ssh key file ${SSH_KEY} does not exist. Exiting."
+    exit 1
+fi
+
+# Collect master provisioning logs
+master_logs_output="${ARTIFACTS}/master_provisioning"
+readonly master_provisioning_logs=(
+    "/tmp/master_extension.log"
+    "/var/log/azure/win-e2e-master-extension.log"
+    "/var/log/azure/cluster-provision.log"
```

Comment:
Since CSE has [`set -x`](https://github.com/Azure/aks-engine/blob/master/parts/k8s/cloud-init/artifacts/cse_main.sh#L4), it may contain sensitive information like service principal credentials (e.g. https://github.com/Azure/aks-engine/blob/master/parts/k8s/cloud-init/artifacts/cse_main.sh#L118-L120). We will have to remove from the log, or execute `set +x` during `docker login` then `set -x`. WDYT?

Reply:
Hmm, indeed. Let me take a better look over all the output from cluster-provision.log and check for all places where sensitive information may be logged. We can easily strip those parts from the log when we collect it. Of course, set +x -x around the docker login command would be a good idea in general for aks-engine since I guess this is not/ won't be the only log collection scenario out there where info like this might best not shared.

Reply:
I propose we merge this without collecting that file for now so we still have some logs, and we'll come back with a solution for the sensitive info.

Reply:
SGTM

Reply:
Pushed with that file commented out.